### ISSUE TITLE
[Dev] MSW not completely initialized by the time bootstrap is called

### ISSUE THREAD

**Issue by kleinfreund** [kumahq/kuma-gui#952]
### What happened?

In an environment with mocks, it’s possible that we make network requests before the initialization of MSW is complete. This seems to be caused by not awaiting the promise returned by `worker.start` in src/services/development.ts. A fix for this isn’t entirely trivial because currently, we can’t register an asynchronous service in our DI layer.

To illustrate the gap, add log statements:

- before `await bootstrap()` in `src/main.ts:25`
- before `worker.start` in `src/services/development.ts:89`
- once `worker.start` resolves (i.e. `.then(() => console.log('worker.start: resolved'))`).

The log before bootstrap will appear before the worker.start call resolves, but it should appear after.

**Comment by johncowen** [comment]
Just to provide an update here:

> In an environment with mocks, it’s possible that we make network requests before the initialization of MSW is complete.

Whilst its technically "possible", its pretty unlikely it seems. The issue that surfaced this realization wasn't actually caused by this at all, more that we were making HTTP requests before we were even calling bootstrap (see above GH links on the issue), and we are yet to see MSW initialization causes issues. That doesn't mean we shouldn't fix this up, just its impact is way smaller than originally thought.

> A fix for this isn’t entirely trivial

This also isn't strictly true. The change is a couple of lines of code, but we are waiting to push something up whilst we see if the actual fix for the issue mentioned above also fixes a flake-y test we have.

>  because currently, we can’t register an asynchronous service in our DI layer.

Additionally `brandi` supports asynchronous service registration and retrieval, but I specifically avoided and would like to continue to avoid using it so that all our `use*` functions follow the same synchronous interface, therefore our little layer over the top of brandi doesn't support doing that and will continue to not support that on purpose unless we have good reason to change.

**Comment by johncowen** [comment]
Bit more information here. I tried putting a `fetch` call right after the call to `worker.start` and MSW catches the request. There is nowhere earlier I can make a `fetch` request after calling `worker.start`.

`worker.start` _does_ resolve _after_ the call to `fetch` but MSW still manages to catch the call to `fetch` even with that being the case.

TLDR; If we amend this or not, it will have zero impact and it currently has zero impact.

I also looked to see how long it has been "technically incorrect" like this and its been like this for years with no impact.
